vpr/src/route/bucket.cpp

```diff
@@ -4,6 +4,27 @@
 #include "vtr_log.h"
 #include "vpr_error.h"
 
+// Initial bucket scaling.  A larger division scaling results in smaller cost
```

Comment:
Either here or later it seems like you need an overview comment on how the heap number of buckets and spacing between buckets is controlled. This describes the individual variables, but I still find division a bit mysterious.  I get the basic idea (I think) of you initially have 100k buckets (or you can grow to 100k buckets) but if the front bucket gets too full you'll increase the number of buckets by 2x and reduce their spacing, until you hit kMaxMaxBuckets.  But there is a bit more to what you're doing than that, and I think an overview comment would help.

Reply:
Big comment written to address a couple of your comments.